addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -0,0 +1,367 @@
+            [
+                ("picking_type_id", "=", picking_type.id),
+                ("location_id", "=", src_loc.id),
+                ("location_dest_id", "=", dest_loc.id),
+                ("group_id", "=", group.id),
+                # NB: only workable pickings
+                ("state", "in", ["assigned", "confirmed", "waiting"]),
+            ]
+        )
+
+        if not picking:
+            # Otherwise reuse the existing picking if all the moves
+            # already belong to it and it contains no other moves
+            # The picking_type_id, location_id and location_dest_id
+            # will match already
+            current_picking = moves.mapped("picking_id")
+            if len(current_picking) == 1 and current_picking.mapped("move_lines") == moves:
+                values = {"group_id": group.id}
+                values.update(kwargs)
+
+                current_picking.write(values)
+                picking = current_picking
+
+        if not picking or len(picking) > 1:
+            # There was no suitable picking to reuse.
+            # Create a new picking.
+            values = {
+                "picking_type_id": picking_type.id,
+                "location_id": src_loc.id,
+                "location_dest_id": dest_loc.id,
+                "group_id": group.id,
+            }
+            values.update(kwargs)
+
+            picking = self.create(values)
+
+        else:
+            # Avoid misleading values for extra fields.
+            # If any of the fields in kwargs is set and its value is different
+            # than the new one, set the field value to False to avoid misleading
+            # values.
+            # For instance, picking.origin is 'ASN001' and kwargs contains
+            # origin with value 'ASN002', picking.origin is set to False.
+            for field, value in kwargs.items():
+                current_value = getattr(picking, field, None)
+                if isinstance(current_value, models.BaseModel):
+                    current_value = current_value.id
+                if current_value and current_value != value:
+                    setattr(picking, field, False)
+
+        moves.write({"group_id": group.id, "picking_id": picking.id})
+
+        move_lines = moves.mapped("move_line_ids")
+        if move_lines:
+            move_lines.write({"picking_id": picking.id})
+            # After moving move lines check entire packages again just in case
+            # some of the move lines are completing packages
+            if picking.state != "done":
+                picking._check_entire_pack()
+
+        return picking
+
+    def _get_package_search_domain(self, package):
+        """Generate the domain for searching pickings of a package"""
+        return [
+            "|",
+            ("move_line_ids.package_id", "child_of", package.id),
+            ("move_line_ids.result_package_id", "child_of", package.id),
+        ]
+
+    def get_pickings(
```

Comment:
Don't think this is really necessary.

Reply:
Was only being used by ```_compute_related_picking_ids``` which needs to go into another module anyway so will remove this.

Reply:
This has now been removed.